Bring back support for grok patterns containing named groups with underscore

kmerz opened the pull request:
Grok Patterns with named groups (`(?<name0>test)`) can contain a name with a underscore (`(?<my_name>test)`).

The java-grok library dropped the support unintentionally by moving the regex Engine from "named-regexp" to java (https://github.com/thekrakken/java-grok/pull/53).

To preserve the support this PR switches back from java to "named-regexp".

kmerz committed:
Bring back pom.xml

kmerz committed:
Bring back 'named-regexp' dependency

kmerz committed:
Replace java.util.regex.* with com.google.code.regexp.*

kmerz committed:
Add test to ensure named underscores work in the future

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/graylog-labs/java-grok?pullRequest=2) <br/>All committers have signed the CLA.

bernd commented on src/main/java/io/krakens/grok/api/Grok.java:
```diff
@@ -5,11 +5,12 @@
 
+import com.google.code.regexp.Matcher;
+import com.google.code.regexp.Pattern;
+
```
The commit that changed to `java.util.regex` in the upstream repository (https://github.com/thekrakken/java-grok/commit/bd098c3a0d76bf9ba2b5b239e6cba34edb8fb82b) changed the following in this file:

![image](https://user-images.githubusercontent.com/461/54839787-e7a60500-4ccb-11e9-918a-554abeb92e1c.png)

With your change we are still calling `GrokUtils.namedGroups(m, m.group());` instead of the previous `m.namedGroups()`. Is that correct or do we need to use `m.namedGroups()` again?

bernd commented on src/main/java/io/krakens/grok/api/Match.java:
```diff
@@ -8,7 +8,9 @@
-import java.util.regex.Matcher;
+
+import com.google.code.regexp.Matcher;
+
```
Same as above, do we need to use `m.namedGroups()` here?

```diff
-    Map<String, String> mappedw = this.match.namedGroups();
+    Map<String, String> mappedw = GrokUtils.namedGroups(this.match,this.subject);
```

kmerz committed:
Fix test8 in BasicTest

The test originally assumed that name0 is
the result with underscore.

I have no idea why I thought this.

The test now will test that a pattern with
underscore will be returned proper after
compilation

kmerz committed:
Revert usage of namedGroups from Grok and Match

kmerz committed:
Fix namedGroups field of Grok.java
